Mathlib/Data/Finset/NatAntidiagonal.lean

```diff
@@ -38,6 +44,12 @@ theorem mem_antidiagonal {n : ℕ} {x : ℕ × ℕ} : x ∈ antidiagonal n ↔ x
   rw [antidiagonal, mem_def, Multiset.Nat.mem_antidiagonal]
 #align finset.nat.mem_antidiagonal Finset.Nat.mem_antidiagonal
 
+/-- `Finset.Nat.antidiagonal` coincides with `Finset.antidiagonal` -/
+lemma antidiagonal_eq_antidiagonal (n : ℕ) :
+    Finset.Nat.antidiagonal n = Finset.antidiagonalOfLocallyFinite.antidiagonal n := by
```

Comment:
This should instead be:

* A `Subsingleton (HasAntidiagonal u)` instance
* Replacing `nat.antidiagonal` with `instance : HasAntidiagonal ℕ`

Reply:
I did that, but I have two questions :
* for the moment, this instance (and the one for Finsupp.HasAntidiagonal) are in their respective files ; maybe they should belong to Mathlib.Data.Finset.Antidiagonal
* My use of the Subsingleton instance is probably awkward
*

Reply:
I think they're best in their respective files.